CMakeLists.txt

```diff
@@ -2752,6 +2753,24 @@ elseif(N3DS)
   sdl_glob_sources("${SDL3_SOURCE_DIR}/src/file/n3ds/*.c")
 endif()
 
+if (SDL_DIALOG)
+  if(UNIX AND NOT APPLE AND NOT RISCOS AND NOT HAIKU)
+    sdl_sources(${SDL3_SOURCE_DIR}/src/dialog/unix/SDL_dialog.c)
+    sdl_sources(${SDL3_SOURCE_DIR}/src/dialog/unix/portal/SDL_dialog.c)
```

Comment:
These files should not all have the same name. Normally the files are prefixed with the appropriate platform or subsystem, e.g. SDL_portaldialog.c, SDL_cocoadialog.m, etc.